iree/turbine/kernel/wave/compile.py

```diff
@@ -0,0 +1,126 @@
+from typing import Any, List, Dict
+from .._support.indexing import IndexingContext, IndexExpr
+from ..compiler import kernel_codegen, host_codegen
+from .compile_options import WaveCompileOptions
+
+from .cache import (
+    is_cache_enabled,
+    get_cache_manager,
+    WAVE_RUNTIME_DIR,
+)
+from .utils.compile_utils import compile_to_vmfb
+from .utils.run_utils import invoke_vmfb, _write_file
+from iree.turbine.kernel._support.context import push, pop
+
+
+class WaveKernel:
+    """
+    Represents a wave kernel that can be invoked by the user.
+    """
+
+    def __init__(self, options: WaveCompileOptions, executable: Any, asm: str):
+        self.options = options
+        self.cached_kernel_hash = self.options.kernel_hash
+        self.subs = options.subs
```

Comment:
Is  `cached_kernel_hash` and `subs` used anywhere?

Reply:
>cached_kernel_hash 
+1,  kernel_hash itself is useful for users to be able to specify kernel hash they want, with that and LRU cache and they can re-use the kernel efficiently without recompiling and recomputing hash at all.  I think within the scope of this class may not be very useful though

Reply:
yeah not being used, will remove. Could be useful to keep track of the hash in the kernel, but can't think of a use case yet. We can bring it back when required.